src/test/java/com/synopsys/integration/stepworkflow/StepWorkflowTest.java

```diff
@@ -35,24 +74,191 @@ public void testSuccessfulSingleStepDatalessStepWorkflow() {
 
     @Test
     public void testUnsuccessfulSingleStepStepWorkflow() {
-        StepWorkflowResponse<Integer> response = StepWorkflow.just(SubStep.ofSupplier(this::unsuccessfulDataSupplier))
+        StepWorkflowResponse<Integer> response = StepWorkflow.just(subStepSupplierException)
                                                      .run();
 
         assertFalse(response.wasSuccessful());
         assertNull(response.getData());
         assertNotNull(response.getException());
```

Comment:
Asserting that the exception is the exception that you expect is likely an even better test here.

Reply:
This is one of the original tests. I didn't review how these were put together after I had implemented additional functionality/tests. I will review and update.